### PR TITLE
fix(components): [anchor] scroll whether link is selected at the top

### LINKED ISSUE #18062: [Component] [anchor, dialog] Anchor 锚点默认选中第一个锚点未生效，而且继续上拉选中消失
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.21`
- Element Plus Version: `2.7.6`
- Browser / OS: `谷歌`
- Build Tool: `Vite`

### Related Component

- `el-anchor`
- `el-dialog`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgcmVhY3RpdmUsIG9uTW91bnRlZCwgbmV4dFRpY2sgfSBmcm9tICd2dWUnO1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBlcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuY29uc3QgY29udGFpbmVyUmVmID0gcmVmKG51bGwpO1xuY29uc3QgYW5jaG9yID0gcmVmKG51bGwpXG5jb25zdCBjdXJyZW50TGluayA9IHJlZignI3BhcnQxJyk7IC8vIOm7mOiupOmAieS4reesrOS4gOS4qumTvuaOpVxuY29uc3QgZnVsbHNjcmVlbj1yZWYoZmFsc2UpXG5jb25zdCBsaW5rcyA9IHJlYWN0aXZlKFtcbiAgeyBocmVmOiAnI3BhcnQxJywgdGl0bGU6ICfllYblk4Hnu4Tln7rmnKzkv6Hmga8nIH0sXG4gIHsgaHJlZjogJyNwYXJ0MicsIHRpdGxlOiAn5ZWG5ZOB57uE6K+m5oOFJyB9XG5dKTtcbmNvbnN0IGhhbmRsZUNsaWNrID0gKGUpID0+IHtcbiAgZS5wcmV2ZW50RGVmYXVsdCgpO1xufTtcbm9uTW91bnRlZCgoKSA9PiB7XG4gIG5leHRUaWNrKCgpID0+IHtcbiAgICBjb25zb2xlLmxvZygnY3VycmVudExpbms6JywgY3VycmVudExpbmsudmFsdWUpO1xuICAgIGNvbnNvbGUubG9nKCdjb250YWluZXJSZWY6JywgY29udGFpbmVyUmVmLnZhbHVlKTtcbiAgICBjdXJyZW50TGluay52YWx1ZSA9ICcjcGFydDEnOyAvLyDorr7nva7mv4DmtLvnmoTpk77mjqVcblxuICAgIC8vIOa7muWKqOWIsOesrOS4gOS4qumUmueCuVxuICAgIGNvbnN0IGZpcnN0QW5jaG9yID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQoJ3BhcnQxJyk7XG4gICAgY29uc29sZS5sb2coJ2ZpcnN0QW5jaG9yOicsZmlyc3RBbmNob3IpXG4gICAgaWYgKGZpcnN0QW5jaG9yKSB7XG4gICAgICBmaXJzdEFuY2hvci5zY3JvbGxJbnRvVmlldyh7IGJlaGF2aW9yOiAnc21vb3RoJyB9KTtcbiAgICB9XG4gIH0pO1xufSk7XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICAgPGVsLXJvdyBjbGFzcz1cImVsLXJvd1wiPlxuICAgICAgPGVsLWNvbCA6c3Bhbj1cIjRcIiBjbGFzcz1cImVsLXJvdy1sZWZ0XCI+XG4gICAgICAgIDxlbC1hbmNob3IgOmNvbnRhaW5lcj1cImNvbnRhaW5lclJlZlwiIGRpcmVjdGlvbj1cInZlcnRpY2FsXCIgdHlwZT1cImRlZmF1bHRcIiBAY2xpY2s9XCJoYW5kbGVDbGlja1wiIDpvZmZzZXQ9XCIzMFwiIDpjdXJyZW50LWluZGV4PVwiMFwiIDpjdXJyZW50LWxpbms9XCJjdXJyZW50TGlua1wiIGJvdW5kPVwiMTBcIj5cbiAgICAgICAgICA8ZWwtYW5jaG9yLWxpbmsgdi1mb3I9XCJsaW5rIGluIGxpbmtzXCIgOmtleT1cImxpbmsuaHJlZlwiIDpocmVmPVwibGluay5ocmVmXCIgOnRpdGxlPVwibGluay50aXRsZVwiIDpjbGFzcz1cInsgJ2lzLWFjdGl2ZSc6IGN1cnJlbnRMaW5rID09PSBsaW5rLmhyZWYgfVwiPlxuICAgICAgICAgIDwvZWwtYW5jaG9yLWxpbms+XG4gICAgICAgIDwvZWwtYW5jaG9yPlxuICAgICAgPC9lbC1jb2w+XG4gICAgICA8ZWwtY29sIDpzcGFuPVwiMjBcIj5cbiAgICAgICAgPGRpdiByZWY9XCJjb250YWluZXJSZWZcIiBjbGFzcz1cInJpZ2h0LXNjcm9sbFwiIDpzdHlsZT1cImZ1bGxzY3JlZW4gPyAnaGVpZ2h0OiA4NXZoOyBvdmVyZmxvdzogc2Nyb2xsOycgOiAnaGVpZ2h0OiA2NXZoOyBvdmVyZmxvdzogc2Nyb2xsOydcIj5cbiAgICAgICAgICA8ZGl2IGlkPVwicGFydDFcIj5cbiAgICAgICAgICAgIDxkaXYgY2xhc3M9XCJib3gtdGl0bGVcIj7llYblk4Hnu4Tln7rmnKzkv6Hmga88L2Rpdj5cbiAgICAgICAgICAgPGRpdiBzdHlsZT0nd2lkdGg6MTAwJTtoZWlnaHQ6NDAwcHg7Jz48L2Rpdj5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgICA8ZWwtZGl2aWRlciAvPlxuICAgICAgICAgIDxkaXYgaWQ9XCJwYXJ0MlwiPlxuICAgICAgICAgICAgPGRpdiBjbGFzcz1cImJveC10aXRsZVwiPuWVhuWTgee7hOivpuaDhTwvZGl2PlxuICAgICAgICAgICAgPGRpdiBzdHlsZT0nd2lkdGg6MTAwJTtoZWlnaHQ6NDAwcHg7Jz48L2Rpdj5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgPC9kaXY+XG4gICAgICA8L2VsLWNvbD5cbiAgICA8L2VsLXJvdz5cbjwvdGVtcGxhdGU+XG5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

直接进入页面下拉才会出现

### What is Expected?

进入组件默认选中第一项，继续上拉选中不会消失

### What is actually happening?

进入组件未默认选中第一项，继续上拉选中消失

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by kcmeven**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix #18062

### CODE REVIEW

**Comment by k983551019** [comment]
https://github.com/element-plus/element-plus/discussions/17977

File: packages/components/anchor/src/anchor.vue [hunk 1]
```diff
@@ -134,13 +134,12 @@ const getCurrentHref = () => {
...
   anchorTopList.sort((prev, next) => prev.top - next.top)
-
   for (let i = 0; i < anchorTopList.length; i++) {
     const item = anchorTopList[i]
     const next = anchorTopList[i + 1]
 
     if (i === 0 && scrollTop === 0) {
-      return ''
+      return props.scrollTop ? item.href : ''
     }
```
Comment (tolking) [inline packages/components/anchor/src/anchor.vue:143]: I found that the problem was caused by this code, and it looked like redundant code. WDYT @Fuphoenixes

**Review by Fuphoenixes — APPROVED** [review]

**Comment by zhangXiaolizi** [comment]
这个merge的内容什么时候发布啊？ 我看最新的版本源码还是没有更新

**Comment by warmthsea** [comment]
> [Open in Stackblitz](https://pkg.pr.new/template/5a365d67-7755-4a08-9079-09ef5a01e89c)
> 
> ```
> pnpm add https://pkg.pr.new/element-plus/element-plus@18047
> ```
> 
> _commit: [`209e9d2`](https://github.com/element-plus/element-plus/runs/31440493040)_

@zhangXiaolizi You can install this link

File: packages/components/anchor/src/anchor.vue (hunk 1 above)
Comment (btea) [reply packages/components/anchor/src/anchor.vue:143]: What do you mean by redundant code?

**Review by btea — APPROVED** [review]

**Comment by btea** [comment]
I see @Fuphoenixes  has approved this PR,  so I merged this PR to verify the pkg.pr.new 403 issue.

But I noticed that the PR added a prop, but the documents were not synchronized, so I created a new PR. #19293

@tolking It seems that your previous comment has not been processed yet. I am not sure if there is anything else that needs to be processed. If so, it may be handled in a new PR.

File: packages/components/anchor/src/anchor.vue (hunk 1 above)
Comment (tolking) [reply packages/components/anchor/src/anchor.vue:143]: I just do not understand why it is designed this way?

**Pull request merged**